gnomad_qc/v4/create_release/calculate_variant_statistics.py

```diff
@@ -215,65 +272,53 @@ def main(args):
 
-        per_sample_ht.checkpoint(
-            get_per_sample_counts(test=test, data_type=data_type).path,
-            overwrite=overwrite,
-        )
-
-    if args.stratify_and_aggregate:
-        per_sample_ht = get_per_sample_counts(test=test, data_type=data_type).ht()
-        strats = []
-        stratify_dictionary = {
-            "all_variants": True,
-            "pass_filters": pass_filters,
-            "ukb_regions": ukb_regions,
-            "broad_regions": broad_regions,
-            "ukb_and_broad_regions": ukb_regions and broad_regions,
-            "rare_variants": rare_variants,
-            "lof": by_csqs,
-            "missense": by_csqs,
-            "synonymous": by_csqs,
-        }
-
-        for key, item in stratify_dictionary.items():
-            strats.append(key)
-        stratified_agg_ht = compute_stratified_agg_stats(
-            per_sample_ht,
-            data_type=data_type,
-            strats=strats,
+    if data_type == "exomes":
+        logger.info("Calculating per-sample variant statistics for exomes...")
+        mt = get_gnomad_v4_vds(
+            test=test, release_only=True, split=True, chrom="chr22" if test else None
+        ).variant_data
+    else:
+        logger.info("Calculating per-sample variant statistics for genomes...")
+        mt = get_gnomad_v4_genomes_vds(
+            test=test, release_only=True, split=True, chrom="chr22" if test else None
+        ).variant_data
+
+    ht = create_per_sample_counts_ht(
+        mt,
+        release_sites(data_type=data_type).ht(),
+        pass_filters=args.pass_filters,
+        ukb_capture=args.filter_ukb_capture_intervals,
+        broad_capture=args.filter_broad_capture_intervals,
+        by_csqs=args.by_csqs,
+        rare_variants=args.rare_variants,
+        vep_canonical=args.vep_canonical,
+        vep_mane=args.vep_mane,
+    ).checkpoint(hl.utils.new_temp_file("per_sample_counts", "ht"))
+
+    if args.add_aggregate_sample_stats_global:
+        logger.info("Computing aggregate sample statistics...")
+        sample_qc_agg_stats = compute_agg_sample_stats(
+            ht,
+            meta_ht=meta(data_type=data_type).ht(),
             by_ancestry=args.by_ancestry,
         )
+        ht = ht.annotate_globals(sample_qc_agg_stats=sample_qc_agg_stats)
 
-        stratified_agg_ht.checkpoint(
-            get_per_sample_counts(
-                test=test,
-                data_type=data_type,
-                suffix="aggregated_",
-            ).path,
-            overwrite=overwrite,
-        )
+    ht = ht.checkpoint(
+        get_per_sample_counts(
+            test=test, data_type=data_type, suffix=args.custom_suffix
+        ).path,
+        overwrite=overwrite,
+    )
+
+    if args.add_aggregate_sample_stats_global:
+        logger.info("Aggregate sample statistics: %s", hl.eval(ht.sample_qc_agg_stats))
 
 
 if __name__ == "__main__":
+    # TODO: Could make some of these arguments --skip-<filter> instead of --<filter> so
```

Comment:
@matren395 @KoalaQin thoughts on this TODO?

Reply:
I can get onboard with that - I'd imagine we'd be filtering to one of broad_regions or ukb_regions (or both) more often than not. Would we do this for: Broad, UKB, and pass_filters ?